cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -110,6 +110,30 @@ void VhpiImpl::get_sim_precision(int32_t *precision)
     *precision = log10int(femtoseconds) - 15;
 }
 
+const char *VhpiImpl::get_simulator_product()
+{
+    if (m_product.empty()) {
+        vhpiHandleT tool = vhpi_handle(vhpiTool, NULL);
+        if (tool) {
+            m_product = vhpi_get_str(vhpiNameP, tool);
+            vhpi_release_handle(tool);
+        }
+    }
+    return m_product.c_str();
+}
+
+const char *VhpiImpl::get_simulator_version()
+{
+    if (m_version.empty()) {
+        vhpiHandleT tool = vhpi_handle(vhpiTool, NULL);
+        if (tool) {
+            m_version = vhpi_get_str(vhpiToolVersionP, tool);
+            vhpi_release_handle(tool);
+        }
```

Comment:
For consistency with the other impls:
```suggestion
        }
        else {
            m_version = "UNKNOWN";
        }
```